internal/v2/s2av2.go

```diff
@@ -0,0 +1,72 @@
+package s2av2
+
+import (
+	"errors"
+	"context"
+
+	"google.golang.org/grpc/credentials"
+	"github.com/google/s2a-go/internal/v2/tlsconfigstore"
+	"github.com/google/s2a-go/internal/handshaker"
+)
+
+const (
+	s2aSecurityProtocol = "s2av2"
+)
+
+type s2av2TransportCreds struct {
+	info     *credentials.ProtocolInfo
+	config   *tls.Config
+	isClient bool
+}
+
+func NewClientCreds() (*credentials.TransportCredentials) {
+	return &s2av2TransportCreds{
+		info: &credentials.ProtocolInfo{
+			SecurityProtocol: s2aSecurityProtocol,
+		},
+		config: tlsconfigstore.GetTlsConfigurationForClient(),
+		isClient: true,
+	}
+}
+
+func NewServerCreds() *credentials.TransportCredentials {
+	return &s2av2TransportCreds{
+		info: &credentials.ProtocolInfo{
+			SecurityProtocol: s2aSecurityProtocol,
+		},
+		config: tlsconfigstore.GetTlsConfigurationForServer(),
+		isClient: false,
+	}
+}
+
+func (c *s2av2TransportCreds) ClientHandShake(ctx context.Context, serverAuthority string, rawConn net.Conn) (_ net.Conn, _ credentials.AuthInfo, err error) {
+	if !c.isClient {
+		return nil, nil, errors.New("client handshake called using server transport credentials")
+	}
+	// TODO(rmehta19): Create a stub to S2A.
+	// TODO(rmehta19): Offload verification of server cert to S2A.
+	// TODO(rmehta19): Offload signing the client's CertificateVerify message to S2A.
+}
+
+
+func (c *s2av2TransportCreds) ServerHandshake() () {
+	if c.isClient {
+		return nil, nil, errors.New("server handshake called using client transport credentials")
+	}
+	// TODO(rmehta19): Create a stub to S2A.
+	// TODO(rmehta19): Offload verification of server cert to S2A.
+	// TODO(rmehta19): Offload signing the client's CertificateVerify message to S2A.
+}
+
+func (c *s2av2TransportCreds) Info() credentials.ProtocolInfo {
+	return *c.info
+}
+
+func (c * s2av2TransportCreds) Clone() *credentials.TransportCredentials {
+	info := *c.info
+	return &s2aTransportCreds{
+		info: &info,
+		config: c.config.Clone()
+		isClient: c.isClient,
+	}
+}
```

Comment:
Don't forget about the `OverrideServerName` function as well:

https://github.com/google/s2a-go/blob/a941a3f6a219d2131d532a9abe27d7bd40244d35/s2a.go#L263